www/adblock2privoxy/Portfile

```diff
@@ -0,0 +1,141 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                adblock2privoxy
+version             2.0.1
+categories          www haskell
+maintainers         @essandess
+license             GPL-3
+platforms           darwin
+
+description         Convert adblock config files to privoxy format
+long_description    ${description}. \
+            AdBlock Plus browser plugin has great block list files \
+            provided by big community, but it is client software and \
+            cannot work on a server as proxy.  Privoxy proxy has good \
+            potential to block ads at server side, but it experiences \
+            acute shortage of updated block lists.  This software \
+            converts adblock lists to privoxy config files format. \
+            Almost all adblock features are supported including \
+            block/unblock requests (on privoxy) all syntax features \
+            are supported except for regex templates matching host \
+            name hide/unhide page elements (via CSS) all syntax \
+            features are supported all block request options except \
+            for outdated ones: Supported: script, image, stylesheet, \
+            object, xmlhttprequest, object-subrequest, subdocument, \
+            document, elemhide, other, popup, third-party, domain=..., \
+            match-case, donottrack.
+
+homepage        https://github.com/essandess/adblock2privoxy
+master_sites    https://hackage.haskell.org/package/adblock2privoxy-${version}
+
+checksums       rmd160  301ba12cb3c624eaca766400972818b79b52adce \
+                sha256  064f501995eef83146ed49ccf83856faee8ab7e6e1fd70013d8bb1e12a3a1411 \
+                size    42119
+
+depends_build-append \
+                port:stack
+
+depends_run-append \
+                port:nginx \
+                port:privoxy
+
+use_configure   no
+
+build {
+    xinstall -m 0755 -d ${workpath}/.stack
+    # the PATH environment must use /usr/bin/gcc, not ${prefix}/bin/gcc
+    system -W ${worksrcpath} "export PATH=/usr/bin:\${PATH} ; \
+        ${prefix}/bin/stack setup --stack-root ${workpath}/.stack --allow-different-user \
+            && ${prefix}/bin/stack build --stack-root ${workpath}/.stack --allow-different-user \
+            && ${prefix}/bin/stack install --stack-root ${workpath}/.stack --local-bin-path ${worksrcpath}/bin --allow-different-user"
```

Comment:
Out of curiosity, would this actually give us exactly the same binary every time it is built, even when dependencies are being updated upstream? I figure the `stack setup` or `stack build` commands will actually download additional source code, right?

Also, while you're copying the binaries into a different path from what you've used as `--local-bin-path` and that obviously seems to work in this case, are you sure this will in the general case, i.e. are you sure that binaries do not depend on where they are compiled and installed to?

And finally, we should probably move the common things into a stack PortGroup and re-use it, rather than copying the instructions around multiple portfiles.

Reply:
Yes, that what `stack install` is designed to do: consistent, reliable, standalone builds over the long term. The resolver `lts-*` in `stack.yaml` stands for long-term support: https://www.stackage.org/. Stack is good at this build job.

I agree that a portgroup for stack is a great idea—there’s many useful projects built with stack.